monkey/monkey_island/cc/ui/src/components/report-components/zerotrust/scoutsuite/ScoutSuiteDataParser.js

```diff
@@ -3,72 +3,104 @@ export default class ScoutSuiteDataParser {
     this.runResults = runResults
   }
 
-  /*
-  itemPath contains path to a specific value e.g. s3.buckets.da1e7081077ce92.secure_transport_enabled"
-  templatePath contains a template path for resource we would want to display e.g. s3.buckets.id
+  /**
+   *
+   * @param itemPath contains path to a specific value e.g. s3.buckets.da1e7081077ce92.secure_transport_enabled
+   * @param templatePath contains a template path for resource we would want to display e.g. s3.buckets.id
+   * @returns {*[]|*}
    */
   getResourceValue(itemPath, templatePath) {
     let resourcePath = this.fillTemplatePath(itemPath, templatePath);
-    return this.getValueAt(resourcePath);
+    return this.getObjectValueByPath(resourcePath, this.runResults);
   }
 
   fillTemplatePath(itemPath, templatePath) {
     let itemPathArray = itemPath.split('.');
     let templatePathArray = templatePath.split('.');
-    let resourcePathArray = templatePathArray.map((val, i) => {return val === 'id' ? itemPathArray[i] : val})
+    let resourcePathArray = templatePathArray.map((val, i) => {
+      return val === 'id' ? itemPathArray[i] : val
+    })
     return resourcePathArray.join('.');
   }
 
-  getValueAt(path) {
-    return this.getValueAtRecursive(path, this.runResults)
+  /**
+   * Retrieves value from ScoutSuite data object based on path, provided in the rule
+   * @param path E.g. a.id.c.id.e
+   * @param source E.g. {a: {b: {c: {d: {e: [{result1: 'result1'}, {result2: 'result2'}]}}}}}
+   * @returns {*[]|*} E.g. ['result1', 'result2']
+   */
+  getObjectValueByPath(path, source) {
```

Comment:
This is definitely better than the single big function from before, but the logic here still seems complex.
Let's review this together later

Reply:
What this code could use is unit tests. I don't care about optimizing the logic here, as it probably won't have to change and is pretty encapsulated. To make this more readable I added comments to every single method here.